lib/services/local/integrations.go

```diff
@@ -196,41 +194,32 @@ func integrationReferencedByEAS(ctx context.Context, bk backend.Backend, name st
 }
 
 // integrationReferencedByAWSICPlugin returns an error if the integration name is referenced
-// by an existing AWS Identity Center plugin. In case the AWS Identity Center plugin exists
-// but does not reference this integration, a conditional action is returned with a revision
-// of the plugin to ensure that plugin hasn't changed during deletion of the AWS OIDC integration.
-func integrationReferencedByAWSICPlugin(ctx context.Context, bk backend.Backend, name string) ([]backend.ConditionalAction, error) {
-	var conditionalActions []backend.ConditionalAction
+// by an existing AWS Identity Center plugin.
+func integrationReferencedByAWSICPlugin(ctx context.Context, bk backend.Backend, name string) error {
 	pluginService := NewPluginsService(bk)
 	plugins, err := pluginService.GetPlugins(ctx, false)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return trace.Wrap(err)
 	}
 
 	for _, p := range plugins {
 		pluginV1, ok := p.(*types.PluginV1)
 		if !ok {
 			continue
 		}
-		if pluginV1.GetType() != types.PluginType(types.PluginTypeAWSIdentityCenter) {
-			continue
-		}
-		if awsIC := pluginV1.Spec.GetAwsIc(); awsIC != nil {
-			switch awsIC.IntegrationName {
-			case name:
-				return nil, trace.BadParameter("cannot delete AWS OIDC integration currently referenced by AWS Identity Center integration %q", pluginV1.GetName())
-			default:
-				conditionalActions = append(conditionalActions, backend.ConditionalAction{
-					Key:       backend.NewKey(pluginsPrefix, name),
-					Action:    backend.Nop(),
-					Condition: backend.Revision(pluginV1.GetRevision()),
-				})
-				return conditionalActions, nil
+		switch pluginV1.GetType() {
+		case types.PluginType(types.PluginTypeAWSIdentityCenter):
+			if awsIC := pluginV1.Spec.GetAwsIc(); awsIC != nil {
+				if awsIC.IntegrationName == name {
```

Comment:
>   // IntegrationName is the Teleport OIDC integration used to gain access to the
  // AWS account. May be empty if [CredentialsSource] is `SYSTEM`.
  // DEPRECATED: Use [Credentials] instead. DELETE in Teleport 19+
  string integration_name = 1 [deprecated = true];

I fee like we should switch to https://github.com/gravitational/teleport/blob/master/api/proto/teleport/legacy/types/types.proto#L7369